### DISCUSSION TITLE
🙏技术报告中关于速度评测的一些疑问

### DISCUSSION THREAD

**Discussion by ValueError-jpg** [magicpdf/Magic-PDF#3620] [Q&A]
1. 请问batched parallel processing protocol 具体是什么策略？另外 MinerU2.5 achieves an end-to-end throughput of 2.12 pages/s，这里的端到端是指从图片到最终markdown结果吗？
2. 对于MonkeyOCR、dots.ocr的token吞吐量是如何计算的？尾注3中的方法好像不适用于他们？

<img width="778" height="798" alt="image" src="https://github.com/user-attachments/assets/b0165a3f-d65b-40f6-84b5-8f05a59cb7d9" />

**Comment by ValueError-jpg** [comment]
论文中提到从原始的0.9页/数调优到2.12页/秒的速度。

非常amazing啊，可以分享下下具体的vllm服务参数吗？技术报告中没有提到。

我们最近在做qwenvl的vllm部署，想得到一些专业性的建议。

**Comment by ValueError-jpg** [comment]
期待指导，感谢！

**Reply by dosubot** [reply]
Hi @ValueError-jpg! I'm [Dosu](https://go.dosu.dev/dosubot) and I’m helping the MinerU team.


关于您的问题，整理如下：

1. **Batched Parallel Processing Protocol 策略**  
MinerU2.5 的批量并行协议会根据 GPU 显存动态设置 batch size（默认 384，可通过环境变量调整），将所有 PDF 页面收集后分批处理。每个 batch 内，版面检测、公式识别、表格识别、OCR 等都采用并行批量推理，OCR 还会按语言和分辨率分组。多 GPU 时，采用“多文件并行”策略，每张卡独立处理不同 PDF 文件，适合大批量任务。部分环节如 PDF 渲染、预处理、后处理仍为串行或 CPU-bound，极端小文件建议单卡处理。整个协议支持通过环境变量和 CLI 参数灵活调优，包括 vLLM 的 --data-parallel-size 等参数[参考](https://github.com/opendatalab/MinerU/discussions/2961#discussioncomment-13730828)。

2. **端到端吞吐量定义**  
2.12 pages/s 的端到端吞吐量指的是从 PDF 图片输入到最终结构化输出（可直接用于 Markdown 转换）的全流程，包括所有模型推理和后处理环节，是真正意义上的 pipeline 性能指标[参考](https://github.com/opendatalab/MinerU/discussions/2961#discussioncomment-13730828)。

3. **vLLM 服务参数与部署建议**  
- 推荐参数：  
  - `--gpu-memory-utilization 0.5~0.85`（根据显存调整，A100/H200 可适当提高）
  - `--tensor-parallel-size` 或 `--data-parallel-size`（多卡并行）
  - `--dtype half`（节省显存）
  - `--max-concurrency`（大批量任务建议调高，如 200）
  - `--model` 指定 MinerU2.5 模型路径
- 环境变量如 `VLLM_USE_V1` 控制自定义 logits processor 是否启用，建议保持默认即可[参考](https://github.com/opendatalab/MinerU/pull/3545)。
- Docker 部署时建议使用官方 vllm/vllm-openai:latest 镜像，50 系显卡建议用 blackwell 基础镜像并升级 PyTorch/vLLM 至 nightly 版，避免 CUDA 兼容性问题[参考](https://github.com/opendatalab/MinerU/issues/3582)。
- 多卡部署时，推荐多进程/多 worker 并发，批量上传 PDF，充分利用 GPU 资源。

4. **MonkeyOCR、dots.ocr 的 token 吞吐量计算**  
目前代码库和文档中没有找到 MonkeyOCR、dots.ocr 的 token 吞吐量具体计算方法，也未见尾注3（“有效 token 数 / 总处理时长”）是否适用。推测这类模型的吞吐量评测逻辑与 MinerU2.5 不完全一致，可能需查阅各自的评测脚本或外部文档。如需精确复现，建议对输出 token 数和处理时长自行统计[参考](#)。

如需具体 Docker Compose、K8s 配置或 CLI 示例，可补充说明。



*To reply, just mention [@dosu](https://go.dosu.dev/dosubot?utm_source=github&utm_medium=bot-comment&utm_campaign=github-comment-footer-20260415&utm_content=reply-with-mention&utm_term=opendatalab%2FMinerU).*

---

Docs are dead. Just use [Dosu](https://dosu.dev?utm_source=github&utm_medium=bot-comment&utm_campaign=github-comment-footer-20260415&utm_content=static-docs-tagline&utm_term=opendatalab%2FMinerU).

[![Leave Feedback](https://img.shields.io/badge/Leave%20Feedback-555555?style=flat)](https://app.dosu.dev/response-feedback?utm_source=github&utm_medium=bot-comment&utm_campaign=github-comment-footer-20260415&utm_content=static-docs-feedback&utm_term=opendatalab%2FMinerU&message_id=f9441b4f-149d-420e-9800-9a4a67ced84d) [![Ask Dosu about MinerU](https://img.shields.io/badge/Ask%20Dosu%20about%20MinerU-2f7b3f?style=flat&logo=data%3Aimage%2Fsvg%2Bxml%3Bbase64%2CPHN2ZyB3aWR0aD0iODYiIGhlaWdodD0iODkiIHZpZXdCb3g9IjAgMCA4NiA4OSIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj48cGF0aCBkPSJNNS4yOTIzNiAxMi43OTI4TDE3Ljc1OTMgNi42ODE4OFY3Mi41NjY3TDUuMjkyMzYgODQuMDYxOFYxMi43OTI4WiIgZmlsbD0iI0I0QkI5MSIvPjxwYXRoIGQ9Ik0xOC4yNTc1IDczLjExOTZMNTkuMTMyOSA3Mi43NDhMNTEuNzAxMSA4Mi40MDk1TDI5LjAzMzggODYuMjkxTDYuMjM5NjIgODUuMTU1NEwxOC4yNTc1IDczLjExOTZaIiBmaWxsPSIjNzc4NTYxIi8%2BPHBhdGggZD0iTTE3LjQ5MTYgMy43MzYzM0wzLjU4NTU3IDEyLjcwOTlWODMuNTc5MkMzLjU4NTU3IDg0Ljc1NDIgNC45ODU2MyA4NS4zNjUyIDUuODQ3MDUgODQuNTY2TDE5LjYyOTYgNzEuNzgwMSIgc3Ryb2tlPSJibGFjayIgc3Ryb2tlLXdpZHRoPSI2LjQyODQ0IiBzdHJva2UtbGluZWNhcD0icm91bmQiLz48bWFzayBpZD0iZG9zdS1kLWN1dG91dCIgZmlsbD0id2hpdGUiPjxwYXRoIGZpbGwtcnVsZT0iZXZlbm9kZCIgY2xpcC1ydWxlPSJldmVub2RkIiBkPSJNNDAuNzA0IDAuNTE4MDY2SDE3LjA0MzlWNzYuMjIyMUg0MC43MDRINDIuNTgwNUg0Ny44MDEzQzY4LjcwNjQgNzYuMjIyMSA4NS42NTMzIDU5LjI3NTIgODUuNjUzMyAzOC4zNzAxQzg1LjY1MzMgMTcuNDY1IDY4LjcwNjMgMC41MTgwNjYgNDcuODAxMyAwLjUxODA2Nkg0Mi41ODA1SDQwLjcwNFoiLz48L21hc2s%2BPHBhdGggZmlsbC1ydWxlPSJldmVub2RkIiBjbGlwLXJ1bGU9ImV2ZW5vZGQiIGQ9Ik00MC43MDQgMC41MTgwNjZIMTcuMDQzOVY3Ni4yMjIxSDQwLjcwNEg0Mi41ODA1SDQ3LjgwMTNDNjguNzA2NCA3Ni4yMjIxIDg1LjY1MzMgNTkuMjc1MiA4NS42NTMzIDM4LjM3MDFDODUuNjUzMyAxNy40NjUgNjguNzA2MyAwLjUxODA2NiA0Ny44MDEzIDAuNTE4MDY2SDQyLjU4MDVINDAuNzA0WiIgZmlsbD0iI0YzRjZGMSIvPjxwYXRoIGQ9Ik0xNy4wNDM5IDAuNTE4MDY2Vi02LjU3OTE5SDkuOTQ2NjlWMC41MTgwNjZIMTcuMDQzOVpNMTcuMDQzOSA3Ni4yMjIxSDkuOTQ2NjlWODMuMzE5NEgxNy4wNDM5Vjc2LjIyMjFaTTE3LjA0MzkgNy42MTUzMkg0MC43MDRWLTYuNTc5MTlIMTcuMDQzOVY3LjYxNTMyWk0yNC4xNDEyIDc2LjIyMjFWMC41MTgwNjZIOS45NDY2OVY3Ni4yMjIxSDI0LjE0MTJaTTQwLjcwNCA2OS4xMjQ5SDE3LjA0MzlWODMuMzE5NEg0MC43MDRWNjkuMTI0OVpNNDIuNTgwNSA2OS4xMjQ5SDQwLjcwNFY4My4zMTk0SDQyLjU4MDVWNjkuMTI0OVpNNDcuODAxMyA2OS4xMjQ5SDQyLjU4MDVWODMuMzE5NEg0Ny44MDEzVjY5LjEyNDlaTTc4LjU1NiAzOC4zNzAxQzc4LjU1NiA1NS4zNTU1IDY0Ljc4NjcgNjkuMTI0OSA0Ny44MDEzIDY5LjEyNDlWODMuMzE5NEM3Mi42MjYxIDgzLjMxOTQgOTIuNzUwNSA2My4xOTQ5IDkyLjc1MDUgMzguMzcwMUg3OC41NTZaTTQ3LjgwMTMgNy42MTUzMkM2NC43ODY2IDcuNjE1MzIgNzguNTU2IDIxLjM4NDcgNzguNTU2IDM4LjM3MDFIOTIuNzUwNUM5Mi43NTA1IDEzLjU0NTMgNzIuNjI2IC02LjU3OTE5IDQ3LjgwMTMgLTYuNTc5MTlWNy42MTUzMlpNNDIuNTgwNSA3LjYxNTMySDQ3LjgwMTNWLTYuNTc5MTlINDIuNTgwNVY3LjYxNTMyWk00MC43MDQgNy42MTUzMkg0Mi41ODA1Vi02LjU3OTE5SDQwLjcwNFY3LjYxNTMyWiIgZmlsbD0iYmxhY2siIG1hc2s9InVybCgjZG9zdS1kLWN1dG91dCkiLz48cGF0aCBkPSJNNjguOTIxNSAzNi4wMTM1QzY4LjkyMTUgMzYuMDEzNSA2NS43MzY5IDQ5LjQ3MzggNTEuNDIzMSA0OS40NzM4QzM3LjEwOTMgNDkuNDczOCAzMi41Nzg3IDM3LjM1OTYgMzIuNTc4NyAzNi4wMTM1IiBzdHJva2U9ImJsYWNrIiBzdHJva2Utd2lkdGg9IjcuNjkxNjEiIHN0cm9rZS1saW5lY2FwPSJyb3VuZCIgc3Ryb2tlLWxpbmVqb2luPSJyb3VuZCIvPjxwYXRoIGQ9Ik0wLjM0ODYzMyA4NS40OTQ2QzAuMzQ4NjMzIDg1LjQ5NDYgMjkuNDg1NiA4NS44MzA5IDM0LjgwOSA4NS42OThDNDQuODMzNyA4NS40NDc3IDUxLjI4NzIgODQuNDAyIDU3LjUyNjkgNzguOTcyNEM2Mi44MTI5IDc0LjM3MjcgNzUuMTM0MiA1OS42ODM2IDc1LjEzNDIgNTkuNjgzNiIgc3Ryb2tlPSJibGFjayIgc3Ryb2tlLXdpZHRoPSI2LjE2NDgyIi8%2BPC9zdmc%2B)](https://github.dosu.com/opendatalab/MinerU?utm_source=github&utm_medium=bot-comment&utm_campaign=github-comment-footer-20260415&utm_content=static-docs-ask-repo&utm_term=opendatalab%2FMinerU) [![Share Dosu with your team](https://img.shields.io/badge/Share%20Dosu%20with%20your%20team-1f6feb?style=flat)](https://app.dosu.dev/signup?referrer=openSource&source=github-footer&utm_source=github&utm_medium=bot-comment&utm_campaign=github-comment-footer-20260415&utm_content=static-docs-share-team&utm_term=opendatalab%2FMinerU)
